common/sagas/notifications.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import { takeEvery, put } from 'redux-saga/effects';
+import {delay} from 'redux-saga'
+import {closeNotification} from 'actions/notifications'
+import type {ShowNotificationAction} from 'actions/notifications'
+
+function* handleNotification(action: ShowNotificationAction) {
+    const {duration} = action.payload
+    // show forever
+    if (duration === 0) {
+        return
+    }
+
+    // FIXME
+    yield delay(duration || 5000)
```

Comment:
Logic is correct,  but this will sleep during tests and whatnot it should be wrapped in `call` or something to fix it not really sure my first SAGA ever.